pkg/grpc/grpc.go

```diff
@@ -156,37 +150,44 @@ func (c Client) valConsMap(vals []staking.Validator) (map[string]staking.Validat
 	return vMap, nil
 }
 
-func SigningValidators(ctx context.Context, cfg config.Config) ([]types.Validator, error) {
-	sVals := []types.Validator{}
-
+func SigningValidators(ctx context.Context, cfg config.Config) (sVals []types.Validator, err error) {
 	client, err := NewClient(cfg)
 	if err != nil {
 		log.Error(err.Error())
 
-		return []types.Validator{}, err
+		return []types.Validator{}, endpointError(err.Error())
 	}
 
-	defer client.connClose()
-
 	sInfos, err := client.SignigInfos(ctx)
+
+	defer func() {
+		if tempErr := client.conn.Close(); tempErr != nil {
+			if err != nil {
+				err = endpointError(fmt.Errorf("%w: %s", err, tempErr).Error())
+			} else {
+				err = endpointError(tempErr.Error())
```

Comment:
Convert these to `fmt.Errorf` statements and in the line `30`  call the `endpointError(err)` ?